ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/tools/osb/mbean/ObjStoreBrowser.java

```diff
@@ -198,44 +204,88 @@ private void registerMBeans() {
+     *
+     * @param osTypeClassName {@link StateManager} type class name
+     * @param beanTypeClassName bean class name which makes tooling available on top of the {@code osTypeClassName}
      * @return whether the type was set OK
      */
     public boolean setType(String osTypeClassName, String beanTypeClassName) {
         try {
-            Class cls = Class.forName(osTypeClassName);
+            Class<?> cls = Class.forName(osTypeClassName);
             StateManager sm = (StateManager) cls.getConstructor().newInstance();
             String typeName = canonicalType(sm.type());
 
-            if (typeName == null || typeName.length() == 0)
-                return false;
-
-            osbTypeMap.put(typeName, new OSBTypeHandler(true, osTypeClassName, beanTypeClassName, typeName, null));
-            typeName = typeName.replaceAll("/", File.separator);
-            osbTypeMap.put(typeName, new OSBTypeHandler(true, osTypeClassName, beanTypeClassName, typeName, null));
-
-            return true;
+            return addType(typeName, osTypeClassName, beanTypeClassName);
         } catch (Exception e) {
-            if (tsLogger.logger.isDebugEnabled())
-                tsLogger.logger.debug("Invalid class type in system property ObjStoreBrowserHandlers: " + osTypeClassName);
+            tsLogger.i18NLogger.warn_invalidObjStoreBrowser_type(osTypeClassName, e);
+            return false;
+        }
+    }
 
+    /**
+     * Add the type to be considered by object browser.
+     * See {@link #setType(String, String)}.
+     * For removing the type see {@link #removeType(String)}.
+     *
+     * @param typeName type name that the bean will operate at
+     * @param osTypeClassName {@link StateManager} type class name
+     * @param beanTypeClassName bean class name which makes tooling available on top of the {@code osTypeClassName}
+     * @return whether the type was set OK
+     */
+    public boolean addType(String typeName, String osTypeClassName, String beanTypeClassName) {
+        if (typeName == null || typeName.length() == 0)
             return false;
+
+        osbTypeMap.put(typeName, new OSBTypeHandler(true, osTypeClassName, beanTypeClassName, typeName, null));
+        typeName = typeName.replaceAll("/", File.separator);
+        osbTypeMap.put(typeName, new OSBTypeHandler(true, osTypeClassName, beanTypeClassName, typeName, null));
+        return true;
+    }
+
+    /**
+     * Removing the type name handler from the processing by the ObjStoreBrowser.
+     * The type won't be considered by {@link #probe()} operation and all registered
+     * type MBeans will be deregistered.
+     *
+     * For adding type to processing see {@link #addType(String, String, String)}.
+     *
+     * @param typeName the type name handler that should be de-registered from handling by browser
+     * @return true if the type name was installed and now removed, false otherwise
+     */
+    public boolean removeType(String typeName) {
```

Comment:
I'm not sure we want to allow the instrumentation types to be dynamic unless there is a requirement for it. Typically users create/develop new instrumentation types and then set them on the browser, ie the types are defined up front and not dynamically.

How would you imagine customers might use this feature?

Reply:
The reason why added it here is the WildFly integration. When the LRA service is stopped or redefined there was lack of the "remove" functionality. So from the "customer" perspective, this is used in WildFly integration part on service start and stop.

I do understand the perspective about the types are not considered to be dynamic and WildFly integration should be fine to reinstall only on JVM restart.

Would you think to not adding `removeType` functionality at all? I can remove it from the PR.

Reply:
It is the JMX MBeans that need removing (see method `private void unregisterMBeans()`) .

If the LRA subsystem is removed then:
- would that not involve a reboot (in which case the WFLY subsystem would not register those LRA type handlers); 
- there should be no more new instances to instrument;
- what does the user do with the existing records left in the object store?

Reply:
Sure, no problem, I removed the method.